src/Xml/Reader/Loader/xml_string_loader.php

```diff
@@ -13,14 +13,14 @@
 /**
  * @return Closure(): XMLReader
  */
-function xml_string_loader(string $xml): Closure
+function xml_string_loader(string $xml, ?string $encoding = null, int $flags = 0): Closure
 {
     return static fn (): XMLReader => disallow_issues(
-        static function () use ($xml): XMLReader {
+        static function () use ($xml, $encoding, $flags): XMLReader {
             Assert::notEmpty($xml, 'The provided XML can not be empty!');
 
             return disallow_libxml_false_returns(
-                XMLReader::XML($xml),
+                XMLReader::fromString($xml, $encoding, $flags),
```

Comment:
@nielsdos: I'm wondering : Why doesn't `fromString` accept documentUri as `fromStream` does?

```
public static function fromStream($stream, ?string $encoding = null, int $flags = 0, ?string $documentUri = null): static {}
public static function fromString(string $source, ?string $encoding = null, int $flags = 0): static {}
```

Is this on purpose or an oversight?

Reply:
The new fromString method was made in analogy to the XML method and has the same behaviour. This was mostly done so people on the ML were happy tbh.
Anyway, documentUri only serves 2 purposes: enrich error messages and set the base directory of the document. For XML/fromString the base directory will be automatically set to the current working directory.
It would be possible to add an extra argument, but is there a specific reason you need to do that?

Reply:
I currently don't need it myself, but noticed the difference in API for both methods.

The reasons for having it would be the same as for `XMLDocument::$documentURI`.

A real-world use-case could be to detect relative imports in XSD files for example.
That way, you could use `XMLReader::$baseURI` to decide what to load next.

Having a good descriptive error message would be a bonus by itself as well.

Reply:
I see, I can consider this for PHP 8.5, but PHP 8.4 is feature frozen.